pkg/controller/install/openshift/openshift.go

```diff
@@ -86,6 +87,168 @@ func EnsureOpenshiftIngress(c client.Client, scheme *runtime.Scheme, instance *s
 	return nil
 }
 
+// EnsureConfigMapForCrt ensures crt injection into the configmap
+func EnsureConfigMapForCrt(c client.Client, scheme *runtime.Scheme, instance *servingv1alpha1.Install) error {
+	namespace := instance.GetNamespace()
+	if routeExists, err := kindExists(c, "route", "route.openshift.io/v1", namespace); err != nil {
+		return err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil
+	}
+
+	if err := updateConfigMapForCrt(c, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateConfigMapForCrt(c client.Client, namespace string) error {
+	cmName := "config-service-ca"
+	cm := &v1.ConfigMap{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: namespace}, cm); err != nil {
+		if errors.IsNotFound(err) {
+			// Define a new configmap
+			cm.Name = cmName
+			cm.Annotations = make(map[string]string)
+			cm.Annotations["service.alpha.openshift.io/inject-cabundle"] = "true"
+			cm.Namespace = namespace
+			err = c.Create(context.TODO(), cm)
+			if err != nil {
+				return err
+			}
+			// ConfigMap created successfully
+			return nil
+		}
+		return err
+	}
+
+	/* Check existence of service-ca.crt in the configMap
```

Comment:
Why is this necessary?

Reply:
I have changed logic so this condition no more exist.

I  added that condition to check if at all  CM exist then that condition should check existence of key and update the annotation.